clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -155,6 +155,7 @@ public void testBasicTransaction() throws InterruptedException {
         prepareFindCoordinatorResponse(Errors.NONE, false, FindCoordinatorRequest.CoordinatorType.TRANSACTION, transactionalId);
 
         sender.run(time.milliseconds());  // find coordinator
+        sender.run(time.milliseconds());
```

Comment:
Yes, since we already had the logic in `Sender` to send `FindCoordinator` when needed, I thought we may as well simplify `initTransactions` so that it uses the same logic as the other transactional requests.